### PR TITLE
TG Mining update Part II

### PR DESCRIPTION
**Pull request by Fox-McCloud**
Ports over the latest and greatest changes to TG mining.

Ports: 
- https://github.com/tgstation/-tg-station/pull/13102
- https://github.com/tgstation/-tg-station/pull/13554 (with my own suggestions/fixes)
- https://github.com/tgstation/-tg-station/pull/8925
- Gar Mesons

I recommend reading: https://github.com/tgstation/-tg-station/pull/13102 for an explanation

long end of it:
- Ore values tweaked a bit to encourage hunting of rarer ores, rather than spamming the hell out of plasma.
 - Yes, lore wise, plasma is important, but when miners are focusing on plasma instead of science materials, there's a problem
- Ore turfs have between 1 and 5 ores in them instead of guaranteed 3
- Can't exploit mining turfs by mousing over them
- Bluespace crystals are now a type of ore that can fit in mining satchels and ore boxes
 - No, you can't redeem them for points.
- Added some goodies to the mining vendor
 - silver pickaxe
 - Gar mesons (mesons that are gaudy as hell and function as a minor weapon)
 - brute first aid kit
 - Hivelord stabilizer (makes it so cores don't go inert)
 - Super kinetic accelerator (more range and more damage)
 - Upgraded resonator (5 fields up as opposed to 3)
- Fixes some sprite issue with the ore redemption/mining vendor machines
- Fixes placing ID into a powered off ore redemption machine.

- R&D can now produces the super accelerator, hyper accelerator, and upgraded resonator.
 - Hyper accelerator has even more range than super and can mine 9 turfs at once
- Mech mining scanner now grants see turfs whenever it scans
- caves can't collide with the xenomorph outpost
- gold grubs start with between 1 and 3 ores already in them

RIPLEY Changes
- RIPLEY's drill cooldown is now halved in a low pressure environment
- RIPLEY's movement speed is faster in a low pressure environment

VG Mining Drone Cube Port
- Ports over VG's drone cubes--just use them and they'll spawn a mining drone.
 - This makes it a bit more convenient for the miner to deploy the drone when and where needed as opposed to having to tug the drone to the mining site
- Makes the drone voucher kit into an actual boxed kit of one drone cube+upgraded welding tool
- Vendor now sells drone cubes as opposed to full drones.

Things I didn't port:

- TG's lootcrates
 - Requires mapping changes, anyway
- Implantable hivelord cores 
 - This is going to require an overhaul of our organ system: This will likely come later when @Aurorablade works on that.

### CODE REVIEW

**Comment by Iamgoofball** [comment]
make sure to port VG's lazarus capsules for the real pokemon experience

**Comment by Spacemanspark** [comment]
Holy shit @Iamgoofball thank you

**Comment by Fox-McCloud** [comment]
I'll look into the capsules--for now though, I just ported the mining drone cubes.

**Comment by Fox-McCloud** [comment]
Ah, didn't realize this was change; ported over the RIPLEY's movement speed change, as well (faster in low pressure, drill has lower CD).

**Comment by Spacemanspark** [comment]
Fair enough.
I looked into it as well, and tried my hand, but, uh... /vg/ code is pretty different.

**Comment by Davdichan** [comment]
I'd request not totally nerfing plasma redeem point cost, as science isn't the only department miners should be mining for. Engineers will still want plasteel from time to time, and cargo exports the stuff too Plasma has more uses but is less desirably than gold and silver? Give me a break.

**Comment by Fox-McCloud** [comment]
The points, to a degree, scale with the rarity of the material in question. Iron ore is superrrrr common and easy to find (don't even need a scanner), and therefore, it's worth less (despite being the most "useful" resource, by far, out of them all).

Plasma is nearly as common as iron ore, so it makes sense it would be on the lower end of the scale, points wise---that said, it's still worth a *lot* for how common it is, at 15 points per ore; plasma really only has three purpose: plasteel, R&D, and cargo points (yes, I realize there's niche purposes outside of that, but that's the big three).

File: code/modules/clothing/glasses/glasses.dm [hunk 1]
```diff
@@ -69,6 +69,19 @@
...
+	icon_state = "garm"
+	item_state = "garm"
+	desc = "Do the impossible, see the invisible!"
+	force = 10
+	throwforce = 10
+	throw_speed = 4
+	attack_verb = list("sliced")
+	hitsound = 'sound/weapons/bladeslice.ogg'
+	sharp = 1
+	edge = 1
```
Comment (TheDZD) [inline code/modules/clothing/glasses/glasses.dm]: These are actually an okay melee weapon (slightly better than toolbox), that's hilarious.

**Pull request merged**